### PR TITLE
Make Java11 the new default

### LINKED ISSUE #662: Default to Java 11 instead of 8
**For Confluent Enterprise Support customers, we would strongly advise you to open a Support ticket which will be addressed within your Support contract SLA at ```https://support.confluent.io```**

**Describe the issue**
The default java version is 8, but I think there are significant performance benefits to using java 11, especially with ssl/tls enabled.

### PR DESCRIPTION
**Pull request by erikgb**
# Description

This PR changes the default JDK from Java8 to Java11. As noted in #662, using Java11 over Java8 has "significant performance benefits", and the assumption is that most users will benefit from this change - if they rely on the default Java configuration. And all Confluent container images have now been migrated to Java11 (Azul Zulu) from CP 6.x.

As far as I know, it is only the HDFS Connector that does not support Java11 a.t.m. So special care should be taken if using this connector. This is also the reason I mark this PR as "breaking" and probably requiring a doc update.

We have been using Java11 with cp-ansible for a long time without any problems....

Fixes #662 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Comment by JumaX** [comment]
@erikgb Great work!  Let me look into getting the docs updated to reflect this change.  We are also mid release(testing), so we may wait to merge this until after the release.  Will confer with the rest of the team.

File: roles/confluent.test/molecule/mtls-java8-rhel/molecule.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ provisioner:
         ssl_enabled: true
         ssl_mutual_auth_enabled: true
 
-        redhat_java_package_name: java-11-openjdk
+        redhat_java_package_name: java-1.8.0-openjdk
```
Comment (domenicbove) [inline roles/confluent.test/molecule/mtls-java8-rhel/molecule.yml:98]: Another thing to consider is the dockerfiles we use for molecule are installing java 8, and then cp-ansible will install java 11 on top, which is a waste... I do like that these are keeping java 8 tests around. I'd say the simplest is just to update the dockerfiles to install java 11 and these scenarios can go without a dockerfile or something

Comment (erikgb) [reply roles/confluent.test/molecule/mtls-java8-rhel/molecule.yml:98]: Good catch and nice suggestion! I will try to fix this.

Comment (erikgb) [reply roles/confluent.test/molecule/mtls-java8-rhel/molecule.yml:98]: I have updated the dockerfiles to install Java11 and also the `certificates.yml` playbook - which seems to be used in the prepare phase of multiple scenarios. The non-default Java scenarios (now Java 8) does not seem to have custom dockerfiles. PTAL!

**Comment by domenicbove** [comment]
@erikgb WDYT about releasing this with the next major version? We can get this merged into master today.

**Comment by domenicbove** [comment]
@erikgb Sorry its not going to make the 6.2.0 (which I'm releasing today), I meant it can go into the next one, which would mean you should update the target branch to `master`

**Comment by erikgb** [comment]
> @erikgb Sorry its not going to make the 6.2.0 (which I'm releasing today), I meant it can go into the next one, which would mean you should update the target branch to `master`

@domenicbove Roger that, PR rebased onto master and changed target branch to master.

**Review by domenicbove — APPROVED** [review]
Thanks for all these changes, we are mordernizing!

**Pull request merged**